src/bn_txns.erl

```diff
@@ -45,7 +50,29 @@ init(Args) ->
             init(Args);
         {ok, State} ->
             persistent_term:put(?MODULE, State),
-            {ok, State#state{}}
+            case application:get_env(blockchain, store_historic_balances, false) of
+                true ->
+                    BalancesDir = filename:join(proplists:get_value(base_dir, Args, "data"), ?BALANCES_DB_FILE),
+                    ok = filelib:ensure_dir(BalancesDir),
+                    GlobalOpts = application:get_env(rocksdb, global_opts, []),
+                    DBOptions = [{create_if_missing, true}, {atomic_flush, true}] ++ GlobalOpts,
+                    case rocksdb:open(BalancesDir, DBOptions) of
+                        {ok, BalancesDB} ->
+                            ChainHeightTab = ets:new(chain_height_tab, []),
+                            case ets:insert(ChainHeightTab, {height, 1}) of
+                                true ->
+                                    lager:info("Starting with historic balance storage...", []),
+                                    {ok, State#state{balances=BalancesDB, chain_height_tab=ChainHeightTab}};
```

Comment:
Ets tables are not needed to be reset this way, but like above, don't use an ETS table here. It won't survive a restart which means next time you'll start over again at a lower height. This would break pretty bad if you ever loaded a snapshot on a node with this approach

Reply:
ah yes good point. will update